### PR TITLE
OSX build patches (Do not merge, WIP)

### PR DESCRIPTION
**Pull request by matt-auckland**
OSX builds are broken on my machine (OSX 10.13.5), this PR fixes this.

# Testing

- [ ] Compiles on Windows
- [ ] Compiles on Linux
- [x] Compiles on another OSX
- [x] Builds navcoin-qt on OSX
- [ ] Wallet functions as normal

### CODE REVIEW

File: doc/build-osx.md [hunk 1]
```diff
@@ -13,97 +13,108 @@ When the popup appears, click `Install`.
...
+## Dependencies
 
-    brew install automake berkeley-db4 libtool boost --c++11 miniupnpc openssl pkg-config homebrew/versions/protobuf260 --c++11 qt5 libevent curl
+    brew install automake berkeley-db4 libtool boost --c++11 miniupnpc openssl pkg-config homebrew/core/protobuf260 --c++11 qt5 libevent curl
 
 NOTE: Building with Qt4 is still supported, however, could result in a broken UI. Building with Qt5 is recommended.
 
-Build NavCoin Core
-------------------------
+## Build NavCoin Core
```
Comment (zanuka) [inline doc/build-osx.md]: @matt-auckland should we insert a step here for running `make` from `/depends` ? Also maybe add some notes about what needs to be on user's PATH for proper compilation, QT flags...

File: src/ntpclient.cpp [hunk 2]
```diff
@@ -7,21 +7,18 @@
...
 
-#include <sstream>
 #include <iomanip>
+#include <sstream>
 
 using namespace boost;
 using namespace boost::asio;
 
-bool CNtpClient::getTimestamp(uint64_t &timeRecv)
+bool CNtpClient::getTimestamp(uint64_t& timeRecv)
```
Comment (red010b37) [inline src/ntpclient.cpp]: Why is this line changed?

File: src/ntpclient.cpp [hunk 3]
```diff
@@ -30,74 +27,57 @@ bool CNtpClient::getTimestamp(uint64_t &timeRecv)
...
             struct timeval timeStruct;
 
             timeStruct.tv_sec = GetArg("-ntptimeout", DEFAULT_NTP_TIMEOUT);
             timeStruct.tv_usec = 0;
             FD_ZERO(&fileDescriptorSet);
 
-            int nativeSocket = socket.native();
-
-            FD_SET(nativeSocket,&fileDescriptorSet);
+            #if BOOST_VERSION >= 104700
```
Comment (red010b37) [inline src/ntpclient.cpp]: Why are we now detecting for the boost version

Comment (matt-auckland) [reply src/ntpclient.cpp]: @aguycalled  found this fix for me, I believe it's a fix for a known bug in boost

File: src/ntpclient.cpp (hunk 2 above)
Comment (matt-auckland) [reply src/ntpclient.cpp]: I think a few things could have been messed with by a code formatter that was probably bundled with the C++ extension for VSCode I installed. I can revert these changes if you want

File: doc/build-osx.md (hunk 1 above)
Comment (matt-auckland) [reply doc/build-osx.md]: See step 2 of this section 😛 Lines 31-33

Comment (matt-auckland) [reply doc/build-osx.md]: That said the fact you didn't see it where you expected means it's not well written

**Comment by zanuka** [comment]
@matt-auckland I can now get core to install, configure and compile properly 👍 
I am able to run `./navcoind` from `/src` and also execute `navcoin-cli` commands. I'll continue testing other functionality... awesome work getting this sorted out! 💯

**Comment by craigmacgregor** [comment]
does the qt binary get built? /src/qt/navcoin-qt

**Comment by craigmacgregor** [comment]
I was able to build navcoind, navcoin-cli and navcoin-tx using this branch, but this was always the case for me. I am still unable to build navcoin-qt using the steps in the doc/build_osx.md 

I get this line of output when i run .configure

configure: WARNING: libprotobuf not found; navcoin-qt frontend will not be built

so i guess this is the problem?

**Comment by craigmacgregor** [comment]
trying with these couple of macports libraries:

https://github.com/bitcoin/bitcoin/issues/7782

**Comment by craigmacgregor** [comment]
This file also needs updating. There's lots of references to navcointalk forum which doesnt exist anymore https://github.com/NAVCoin/navcoin-core/blob/osx-build-patches/doc/README.md

File: doc/build-osx.md [hunk 4]
```diff
@@ -13,97 +13,110 @@ When the popup appears, click `Install`.
...
 
-    chown root /usr/local/bin/brew
-    sudo brew update
-    sudo chown -R $USER /usr/local
-    brew install qt@5.7
+```bash
+chown root /usr/local/bin/brew
+sudo brew update
+sudo chown -R $USER /usr/local
+brew install qt@5.7
```
[the remark this answers is not in the document]
Comment (zanuka) [reply doc/build-osx.md]: removing old versions of qt and re-installing w/ just `brew install qt5` did the trick. also there was something odd going on w/ protobuf so reinstalled that as well. 

`./navcoin-qt` now avail 👍

**Comment by matt-auckland** [comment]
Still not sure why Travis is failing, if anyone has thoughts on this I'd appreciate it. Keen to see this merged.

@craigmacgregor I'll update those readme issues you raised in a separate PR. Not sure what's up with the libprotobuf issue

**Comment by matt-auckland** [comment]
I'm going to split this into two branches: 
- Documentation improvements
- Build fixes

No need to hold the documentation improvements back

**Comment by aguycalled** [comment]
ssl-cert of build.nav.community should be fixed now

**Comment by aguycalled** [comment]
should this be closed?

**Comment by craigmacgregor** [comment]
closing unless otherwise re-opened
